src/Groups/abelian_aut.jl

```diff
@@ -0,0 +1,89 @@
+function _isomorphic_gap_group(A::GrpAbFinGen; T=PcGroup)
+  # find independent generators
+  if isdiagonal(rels(A))
+    exponents = [Int(a) for a in diagonal(rels(A))]
+    A2 = A
+    A2_to_A = identity_map(A)
+    A_to_A2 = identity_map(A)
+    else
+    exponents = [Int(a) for a in elementary_divisors(A)]
+    A2, A2_to_A = snf(A)
+    A_to_A2 = inv(A2_to_A)
+  end
+  # the isomorphic gap group
+  Agap = abelian_group(T, exponents)
+  G = GAP.Globals
+  # the gap IndependentGenerators may differ from
+  # the generators even if the generators are independent
+  gensindep = G.IndependentGeneratorsOfAbelianGroup(Agap.X)
+  Aindep = abelian_group([G.Order(g) for g in gensindep])
+
+  imgs = [GAP.gap_to_julia(Vector{fmpz},G.IndependentGeneratorExponents(Agap.X, a.X)) for a in gens(Agap)]
+  A2_to_Aindep = hom(A2, Aindep, [Aindep(e) for e in imgs])
+  Aindep_to_A2 = inv(A2_to_Aindep)
+  Aindep_to_A = compose(Aindep_to_A2, A2_to_A)
+
+  function Agap_to_A(a)
+      return Aindep_to_A(_gap_to_oscar(a, Aindep))
+  end
+
+  function A_to_Agap(a)
+      return _oscar_to_gap(A_to_A2(a), Agap)
+  end
+
+  to_gap = Hecke.map_from_func(A_to_Agap, A, Agap)
+  to_oscar = Hecke.map_from_func(Agap_to_A, Agap, A)
+
+  return Agap, to_gap, to_oscar
+end
+
+function _oscar_to_gap(a::GrpAbFinGenElem, B::GAPGroup)
+  gensB = gens(B)
+  n = length(a.coeff)
+  @assert length(gensB) == n
+  img = one(B)
+  for i in 1:n
+    img *= gensB[i]^a[i]
+  end
+  return img
+end
+
+function _gap_to_oscar(a::Oscar.BasicGAPGroupElem, B::GrpAbFinGen)
+  A = parent(a)
+  G = GAP.Globals
+  exp = GAP.gap_to_julia(Vector{fmpz},G.IndependentGeneratorExponents(A.X, a.X))
```

Comment:
`gap_to_julia` should almost always be avoided. better:
```suggestion
  exp = Vector{fmpz}(GAP.Globals.IndependentGeneratorExponents(A.X, a.X))
```

Reply:
Good to know. Is this documented anywhere?

Reply:
The possibility of doing conversions this way is documented in the GAP.jl documentation, see https://oscar-system.github.io/GAP.jl/dev/conversion/

But the preference for using constructors is not yet in there, this is a relatively recent decision. It makes the code tighter and look more like conventional Julia. That said, `gap_to_julia` with an explicit type is also fine; what we really want to get away from is the old `gap_to_julia(gapobj)` which "guesses" a type.